tests/complex_test.py

```diff
@@ -1,21 +1,94 @@
-from .context import assert_equal
+from .context import _Mul, assert_equal
 import pytest
-from sympy import Sum, I, Symbol, Integer
+import hashlib
+from sympy import Sum, I, Symbol, Integer, Add, Mul, Rational, exp, pi, Pow, re, im, arg, Abs
 
 a = Symbol('a', real=True, positive=True)
 b = Symbol('b', real=True, positive=True)
 i = Symbol('i', real=True, positive=True)
 n = Symbol('n', real=True, positive=True)
 x = Symbol('x', real=True, positive=True)
 
+complex_expression_examples = [
+    ('a+\\imaginaryI b', Add(a, Mul(I, b, evaluate=False), evaluate=False), {}),
+    ('a+\\imaginaryJ b', Add(a, Mul(I, b, evaluate=False), evaluate=False), {}),
+    ('e^{\\imaginaryI \\pi }', exp(_Mul(I, pi), evaluate=False), {}),
+    ('e^{\\imaginaryJ \\pi }', exp(_Mul(I, pi), evaluate=False), {}),
+    ('\\sum_{i=0}^{n} i \\cdot x', Sum(i * x, (i, 0, n)), {}),
+]
 
-def test_complex():
-    assert_equal("a+Ib", a + I * b)
 
+@pytest.mark.parametrize('input, output, variable_values', complex_expression_examples)
+def test_complex_expr(input, output, variable_values):
+    assert_equal(input, output, variable_values)
 
-def test_complex_e():
-    assert_equal("e^{I\\pi}", Integer(-1))
 
+complex_number_rectangular_examples = [
+    ('3+\\imaginaryI 4', Add(3, Mul(I, 4, evaluate=False), evaluate=False), {}),
+    ('-3-4\\imaginaryI ', Add(-3, Mul(-1, 4, I, evaluate=False), evaluate=False), {}),
+    ('3.14+\\imaginaryI 4.657', Add(Rational(157, 50), Mul(I, Rational(4657, 1000), evaluate=False), evaluate=False), {}),
+    ('3+\\imaginaryJ 4', Add(3, Mul(I, 4, evaluate=False), evaluate=False), {}),
+    ('-3-4\\imaginaryJ ', Add(-3, Mul(-1, 4, I, evaluate=False), evaluate=False), {}),
+    ('3.14+\\imaginaryJ 4.657', Add(Rational(157, 50), Mul(I, Rational(4657, 1000), evaluate=False), evaluate=False), {}),
+    ('\\variable{a}+\\imaginaryJ \\variable{b}', Add(Symbol('a' + hashlib.md5('a'.encode()).hexdigest(), real=True), Mul(I, Symbol('b' + hashlib.md5('b'.encode()).hexdigest(), real=True), evaluate=False), evaluate=False), {}),
```

Comment:
@builtbywill is this MD5 stuff just some internal sympy artifact? Or?

Reply:
@jkdufair no that's just how variable symbol names are generated in the latex2sympy code

Reply:
to differentiate them from just letter symbols